File: dayan-server/dayan-modules/dayan-module-agent/src/main/java/com/dayan/agent/entity/LearningContent.java
package com.dayan.agent.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.dayan.common.mybatis.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * 学习中心内容（learning_content）。
 *
 * <p>全局表（无 channel_code），所有代理人可见。
 * 分类：1=视频课程 2=图文课程 3=雁鸣中国。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("learning_content")
public class LearningContent extends BaseEntity {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /** 内容编码 LC+yyyyMMdd+seq */
    private String contentCode;

    /** 标题 */
    private String title;

    /** 摘要 */
    private String summary;

    /** 分类 1=渠道课程 2=外部课程 3=雁鸣中国 */
    private Integer category;

    /** 讲师/作者/来源 */
    private String author;

    /** 时长（视频 28:30 / 图文 约15分钟） */
    private String duration;

    /** 正文（详情页用） */
    private String body;

    /** 阅读/播放量 */
    private Integer viewCount;

    /** 角标（热/新/要闻/人物/动态/洞察） */
    private String badge;

    /** 发布时间 */
    private LocalDateTime publishTime;

    /** 排序权重（越大越靠前） */
    private Integer sortOrder;

    /** 1=上架 0=下架 */
    private Integer status;
}

File: dayan-admin/src/types/learning-content.ts
/** 学习中心板块内容（learning_content，Admin 端） */

/** 板块分类：1=渠道课程 2=外部课程 3=雁鸣中国（大雁课程=course_info，走课程管理首 tab） */
export const LEARNING_CATEGORY_OPTIONS = [
  { label: '渠道课程', value: 1 },
  { label: '外部课程', value: 2 },
  { label: '雁鸣中国', value: 3 }
] as const

export interface LearningContentItem {
  /** 雪花 id，后端序列化为字符串防 JS 精度丢失 */
  id?: string
  contentCode?: string
  title: string
  summary?: string
  category: number
  author?: string
  duration?: string
  body?: string
  viewCount?: number
  badge?: string
  publishTime?: string
  sortOrder?: number
  status?: number
  createdAt?: string
  updatedAt?: string
}

export interface LearningContentQuery {
  current: number
  size: number
  title?: string
  category?: number
  status?: number
}
